shared/provision/forgot-username/index.tsx

```diff
@@ -1,40 +1,49 @@
 import * as React from 'react'
 import * as Kb from '../../common-adapters'
+import * as Constants from '../../constants/provision'
+import * as Container from '../../util/container'
+import * as ProvisionGen from '../../actions/provision-gen'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
 import * as Styles from '../../styles'
+import * as SettingsGen from '../../actions/settings-gen'
 import PhoneInput from '../../signup/phone-number/phone-input'
 import {SignupScreen, errorBanner} from '../../signup/common'
 
-type Props = {
-  forgotUsernameResult: string
-  onBack: () => void
-  onSubmit: (email: string, phone: string) => void
-  waiting: boolean
-}
+const ForgotUsername = () => {
+  const dispatch = Container.useDispatch()
+
+  const defaultCountry = Container.useSelector(state => state.settings.phoneNumbers.defaultCountry)
+  // trigger a default phone number country rpc if it's not already loaded
+  React.useEffect(() => {
+    defaultCountry === '' && dispatch(SettingsGen.createLoadDefaultPhoneNumberCountry())
+  }, [defaultCountry, dispatch])
+
+  const forgotUsernameResult = Container.useSelector(state => state.provision.forgotUsernameResult)
+  const onBack = React.useCallback(() => dispatch(RouteTreeGen.createNavigateUp()), [dispatch])
+  const waiting = Container.useAnyWaiting(Constants.forgotUsernameWaitingKey)
 
-const ForgotUsername = (props: Props) => {
   const [emailSelected, setEmailSelected] = React.useState(true)
   const [email, setEmail] = React.useState('')
   // PhoneInput's callback gets passed a (phoneNumber, valid) tuple.
   // If "valid" is false, phoneNumber gets set to null, therefore phoneNumber is only
   // truthy when it's valid. This is used in the form validation logic in the code.
   const [phoneNumber, setPhoneNumber] = React.useState<string | null>(null)
-  const {onSubmit} = props
-  const _onSubmit = React.useCallback(() => {
+  const onSubmit = React.useCallback(() => {
     if (!emailSelected && phoneNumber) {
-      onSubmit('', phoneNumber)
+      dispatch(ProvisionGen.createForgotUsername({email: '', phone: phoneNumber}))
     } else if (emailSelected) {
-      onSubmit(email, '')
+      dispatch(ProvisionGen.createForgotUsername({email, phone: ''}))
```

Comment:
I know it's not your code but since you're already touching it can you convert this action to use undefined rather than empty string (make empty strings in the action handler when it calls the RPC if you need to)?